router/h2/src/main/scala/io/buoyant/router/H2.scala

```diff
@@ -86,7 +86,7 @@ object H2 extends Router[Request, Response]
     client.newClient(dest, label)
 
   object Server {
-    val newStack: Stack[ServiceFactory[Request, Response]] = FinagleH2.Server.newStack
+    val newStack: Stack[ServiceFactory[Request, Response]] = (ForwardClientCertFilter.module +: FinagleH2.Server.newStack)
```

Comment:
I think it makes more sense to put this module on the client stack instead of the server stack.  That way, sending of the client cert header can be configured per-destination rather then being on/off for the entire router.